packit_service/worker/reporting.py

```diff
@@ -240,7 +254,56 @@ def report_status_by_comment(
     def get_statuses(self):
         self.project_with_commit.get_commit_statuses(commit=self.commit_sha)
 
-    def comment(self, body: str):
+    def _has_identical_comment(self, body: str, mode: DuplicateCheckMode) -> bool:
+        """Checks if the body is the same as the last or any comment based on mode."""
+        if self.pr_id:
+            for comment in self.project.get_pr(pr_id=self.pr_id).get_comments(
+                reverse=True
+            ):
+                if comment.author.startswith(self._packit_user):
+                    if mode == DuplicateCheckMode.check_last_comment:
+                        return body == comment.body
+                    elif (
+                        mode == DuplicateCheckMode.check_all_comments
+                        and body == comment.body
+                    ):
+                        return True
+        else:
+            commit_comments = self.project.get_commit_comments(self.commit_sha)
+            for comment in reversed(commit_comments):
+                if comment.author.startswith(self._packit_user):
+                    if mode == DuplicateCheckMode.check_last_comment:
+                        return body == comment.comment
+                    elif (
+                        mode == DuplicateCheckMode.check_all_comments
+                        and body == comment.comment
+                    ):
+                        return True
```

Comment:
```suggestion
        comments = (
            self.project.get_pr(pr_id=self.pr_id).get_comments(reverse=True)
            if self.pr_id
            else reversed(self.project.get_commit_comments(self.commit_sha))
        )
        for comment in comments:
            if comment.author.startswith(self._packit_user):
                if mode == DuplicateCheckMode.check_last_comment:
                    return body == comment.body
                elif (
                    mode == DuplicateCheckMode.check_all_comments
                    and body == comment.body
                ):
                    return True
```

aaaa /o\ I see it now… I'd just add a property (and ideally deprecate the current one) to the commit comment to unify it, since it's the only one we haven't touched during the ogr refactoring.

It looks like we use `body` for the refactored issue/PR comments and it's also used in other objects (like `Release`)

Reply:
Yes, that's what I wanted to do initially but then realized the different interfaces. I will create an ogr PR to change `comment` to `body`